docs/schema-conversion-logic.md

```diff
@@ -0,0 +1,98 @@
+# Schema conversion logic
+
+Metrictank comes with a schema conversion logic (`mdata/importer/converter.go`) which can be used to convert data from one storage schema into another one. This is currently only used by the whisper importer, but may in the future also be used to import from other types of data sources.
+
+The conversion happens on a destination archive by destination archive basis. In this context the term "archive" refers to one section between the `,` of a storage schema, f.e. the schema `1s:1d,10min:1y` has two archives. Each destination archive gets generated separately, in some cases the same input archives may be used to generate multiple destination archives and in others different input archives may be used. Multiple input archives may also be combined into a single destination archive, if this makes sense given the input schema and the destination schema.
+
+This is how the schema conversion logic generates a destination archive:
+
+  * In the set of source archives which it has received as input, it finds the archive with the shortest retention which still contains enough data to satisfy the retention of the destination archive
```

Comment:
does that mean in this case, that it will try to use the first archive? that seems suboptimal:
```
dest: 15s:1d
src: 1s:1d,15s:5d
```

worse what if the src is `10s:1d,15s:5d` ? It can't convert 10s to 15s

Reply:
I think you just found a bug. I was going through that scenario where `1s:1d,15s:5d` gets converted to `15s:1d`, and it doesn't produce the expected result. let me write a unit test for that and then fix it

Reply:
I'll consider that high prio because we're planning to use the schema conversions in a prod instance soon. I don't think we'll have to do a conversion which hits this bug there, afaict, but i'd still rather get it fixed asap

Reply:
https://github.com/grafana/metrictank/issues/1596

Reply:
not to distract from the bug here, but `10s:1d,15s:5d` isnt a valid retention policy, at least not when using Whisper. https://graphite.readthedocs.io/en/latest/whisper.html#archives-retention-and-precision

> the precision of a longer retention archive must be divisible by precision of next lower retention archive.

Should we also enforce this in Metrictank?

Reply:
good catch -> https://github.com/grafana/metrictank/issues/1599

Reply:
so don't we need to update this text now?